py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py

```diff
@@ -446,24 +446,49 @@ def flip(
     output_shape = list(input.shape)
     stride_slice = []
 
+    dynamic_shape = has_dynamic_shape(input.shape)
+
     shape = input.shape
     rank = len(shape)
     dims = get_positive_dim(dims, rank)
 
     for i in range(rank):
         if i in dims:
-            start_slice.append(shape[i] - 1)
+            if shape[i] == DYNAMIC_DIM:
+                dim = get_shape(ctx, target, source_ir, f"{name}_shape_dim", input, i)
```

Comment:
Since here's inside of `for` loop, please add suffix `i` to the name to avoid duplicated names.